lib/vector-vrl-functions/src/get_metadata_field.rs

```diff
@@ -38,13 +42,16 @@ impl Function for GetMetadataField {
         _ctx: &mut FunctionCompileContext,
         mut arguments: ArgumentList,
     ) -> Compiled {
-        let key = arguments
-            .required_enum("key", &super::keys())?
-            .try_bytes_utf8_lossy()
-            .expect("key not bytes")
-            .to_string();
+        let key_bytes = arguments
+            .required_literal("key")?
+            .to_value()
+            .as_bytes()
+            .cloned()
+            .expect("key not bytes");
+        let key = String::from_utf8_lossy(key_bytes.as_ref());
+        let lookup = compile_path_arg(key.as_ref())?;
```

Comment:
I didn't realize we could take paths as an argument. This will make things a bit nicer, especially backwards compatibility. I'll definitely switch this over.